src/hooks/useApiHook.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from 'react';
+
+export type TApiResponse = {
+  status: number;
+  statusText: string;
+  data: any;
+  error: any;
+  loading: boolean;
+};
+
+export const useApiGet = (url: string): TApiResponse => {
```

Comment:
do you use this hook for fetching movies  ? It should be included in the store. you need to initialize the store with movie data populated